jadiv_code_master.py

```
@@ -1092,14 +1153,25 @@ def launch_app(self, app):
         # Capture stderr to a temp file (not a pipe — a long-running app that
         # logs a lot would eventually fill an unread pipe buffer and hang) so
         # that if the process dies right away (e.g. a missing dependency) we
         # can show *why* instead of silently reporting "Launched" while
         # nothing actually opens.
         log_fd, log_path = tempfile.mkstemp(prefix="codemaster-launch-")
         try:
-            proc = subprocess.Popen(run, shell=True, cwd=str(cwd),
+            # No shell=True: argv is executed directly, so shell
+            # metacharacters in a publisher-supplied run command can't be
+            # used to inject extra commands.
+            proc = subprocess.Popen(argv, cwd=str(cwd),
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)